src/pages/mydashboard/components/SearchDashboard/index.tsx

```diff
@@ -0,0 +1,61 @@
+import styles from './SearchDashboard.module.scss';
+import Image from 'next/image';
+import { Dispatch, SetStateAction, useState } from 'react';
+
+type Invitation = {
+  id: number;
+  inviter: { nickname: string; email: string; id: number };
+  teamId: string;
+  dashboard: { title: string; id: number };
+  invitee: { nickname: string; email: string; id: number };
+  inviteAccepted: boolean;
+  createdAt: string;
+  updatedAt: string;
+};
+
+type SearchDashboardProps = {
+  initialInvitations: Invitation[];
+  setInvitations: Dispatch<SetStateAction<Invitation[]>>;
+};
+
+function SearchDashboard({
+  initialInvitations,
+  setInvitations,
+}: SearchDashboardProps) {
+  const [searchText, setSearchText] = useState('');
+
+  const handleSubmit = () => {
+    const searchInvitations = initialInvitations.filter(
+      (invitation) =>
+        invitation.dashboard.title.includes(searchText) ||
+        invitation.inviter.nickname.includes(searchText),
+    );
+    setInvitations(searchInvitations);
+  };
+
+  const handleKeyEnter = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
+      handleSubmit();
+      return;
+    }
+  };
+
+  return (
+    <div className={styles.searchDashboard}>
+      <Image
+        className={styles.search}
+        src="/svgs/search.svg"
+        alt="검색 아이콘"
+        width={24}
+        height={24}
+      />
+      <input
+        placeholder="검색"
+        onChange={(e) => setSearchText(e.target.value)}
+        onKeyDown={(e) => handleKeyEnter(e)}
```

Comment:
```suggestion
        onKeyDown={(e) => handleKeyEnter(e)}
```
검색 기능이 엔터키 누를때보다 onChange 할 때 나오면 더 좋을 것 같아요!! 검색 api도 호출하시니깐 디바운싱 적용하는 것도 추천드려요
지금은 검색 기능이 클라이언트에서 돌아가고 있는 것 같네요? 이것도 title로 넘겨주셔도 좋을 것 같아요!!